sunpy/util/decorators.py

```diff
@@ -264,3 +266,47 @@ def __call__(self, func):
         if self.kwargs:
             func.__doc__ = func.__doc__.format(**self.kwargs)
         return func
+
+
+def deprecate_positional_args(f):
+    """
+    Decorator for methods that issues warnings for positional arguments
+    Using the keyword-only argument syntax in pep 3102, arguments after the
+    * will issue a warning when passed as a positional argument.
+
+    Parameters
+    ----------
+    f : function
+        Function to check arguments on.
+
+    References
+    ----------
+    Taken from from `scikit-learn <https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L1271>`__.
+    Licensed under the BSD, see "licenses/SCIKIT-LEARN.rst".
+    """
+    sig = signature(f)
+    kwonly_args = []
+    all_args = []
+
+    for name, param in sig.parameters.items():
+        if param.kind == Parameter.POSITIONAL_OR_KEYWORD:
+            all_args.append(name)
+        elif param.kind == Parameter.KEYWORD_ONLY:
+            kwonly_args.append(name)
+
+    @wraps(f)
+    def inner_f(*args, **kwargs):
+        extra_args = len(args) - len(all_args)
+        if extra_args > 0:
+            # ignore first 'self' argument for instance methods
+            args_msg = ['{}={}'.format(name, arg)
```

Comment:
```suggestion
            args_msg = [f'{name}={arg}'
```
Where are the f-string police? =P